builtin/providers/clc/resource_clc_load_balancer.go

```diff
@@ -0,0 +1,124 @@
+package clc
+
+import (
+	"fmt"
+	"log"
+
+	clc "github.com/CenturyLinkCloud/clc-sdk"
+	"github.com/CenturyLinkCloud/clc-sdk/lb"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCLCLoadBalancer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCLCLoadBalancerCreate,
+		Read:   resourceCLCLoadBalancerRead,
+		Update: resourceCLCLoadBalancerUpdate,
+		Delete: resourceCLCLoadBalancerDelete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"data_center": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// optional
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// computed
+			"ip_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCLCLoadBalancerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clc.Client)
+	dc := d.Get("data_center").(string)
+	name := d.Get("name").(string)
+	desc := d.Get("description").(string)
+	status := d.Get("status").(string)
+	r1 := lb.LoadBalancer{
+		Name:        name,
+		Description: desc,
+		Status:      status,
+	}
+	l, err := client.LB.Create(dc, r1)
+	if err != nil {
+		return fmt.Errorf("Failed creating load balancer under %v/%v: %v", dc, name, err)
+	}
+	d.SetId(l.ID)
+	// platform bug - race condition: poll for non-404
+	for {
+		_, err := client.LB.Get(dc, l.ID)
+		if err == nil {
+			break
+		}
+	}
```

Comment:
Let's avoid the infinite loop here - we have a helper for this that makes it easier - check out [`resource.Retry`](https://github.com/hashicorp/terraform/blob/f09a5661cfec1239615a67d1143493dd7fa1b5f2/builtin/providers/aws/resource_aws_vpc.go#L311-L332).